ddtrace/internal/dm.py

```diff
@@ -0,0 +1,100 @@
+import enum
+from hashlib import sha256
+import re
+from typing import TYPE_CHECKING
+
+from ddtrace import config
+from ddtrace.internal.constants import DEFAULT_SERVICE_NAME
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+if TYPE_CHECKING:
+    from typing import Optional
+    from typing import Text
+    from typing import Union
+
+    from ddtrace.context import Context
+
+
+class SamplingMechanism(enum.Enum):
```

Comment:
I think we're moving away from using Enums for performance reasons. SamplingMechanism could inherit from object and have a list of constants.